### PR TITLE
Fix initialization sequence (bug introduce by b4dd323)

### PR DESCRIPTION
**Pull request by rohgarg**
This is a port of @karya0's PR #372  for 3.0 to the 2.5 branch.

### CODE REVIEW

**Comment by rohgarg** [comment]
@balajir1990: This patch should fix the hang-up issue for you.

File: src/dmtcp_coordinator.cpp [hunk 1]
```diff
@@ -298,13 +298,31 @@ void DmtcpCoordinator::handleUserCommand(char cmd, DmtcpMessage* reply /*= NULL*
...
+      if (clientHosts.find(clients[i]->hostname()) == clientHosts.end()) {
+        clientHosts[clients[i]->hostname()] = 1;
+      } else {
+        clientHosts[clients[i]->hostname()] += 1;
+      }
+    }
+    for (dmtcp::map<string,int>::iterator it=clientHosts.begin(); it!=clientHosts.end(); ++it)
+      JASSERT_STDERR << it->first << " => " << it->second << '\n';
     break;
+  }
```
Comment (karya0) [inline src/dmtcp_coordinator.cpp]: I don't think thus block belongs to this commit.

File: src/dmtcpworker.h [hunk 2]
```diff
@@ -44,6 +44,7 @@ namespace dmtcp
 
       static void waitForCoordinatorMsg(string signalStr,
                                         DmtcpMessageType type);
+      static void initialize();
```
Comment (karya0) [inline src/dmtcpworker.h]: Stale?

File: src/syscallsreal.c [hunk 3]
```diff
@@ -305,14 +305,15 @@ void dmtcp_prepare_wrappers(void)
   }
 }
 
+void dmtcp_initialize();
```
Comment (karya0) [inline src/syscallsreal.c]: We can remove this from here.

**Comment by rohgarg** [comment]
@karya0: I have now addressed your comments. Let me know if this is good to go.

**Pull request closed without merging**